### DISCUSSION TITLE
How do I skip the 'reset' sort state for a column?

### DISCUSSION THREAD

**Discussion by yyyyaaa** [TanStack/table#3559] [Q&A]
When using `useSortBy` hook, each column will have 3 sort state: asc, desc, reset. How do I skip the 'reset' state and default to initially either asc or desc? I can't find anything in the docs that mention this or any relevant API.

**Comment by kathesama** [comment]
just add disableSortRemove: true to useTable(),
ex:
`} = useTable((
    {
      columns,
      data,
      disableSortRemove: true,
    },`

**Comment by CatPerry** [comment]
For anyone coming to this React Table sorting question later, here's a bit more specificity. If you use `disableSortRemove: true` alone you'll still have the very first click be an "unsorted" state, and then subsequent clicks are sorted asc or desc. That's perhaps not the UI you want, as a button click, first or not, should do something, but with that one variable, on first click it wont. 

So that you dont have to click the sort button twice to see sorting, do this!!
```
const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable(
    {
      columns: tableColumns,
      data: tableData,
      disableSortRemove: true,
      defaultCanSort: true,
      initialState: {
        sortBy: [{ id: tableData[0]?.totalSeries?.totalSeries, desc: true }], //TODO: adjust TableData type file
      } as any,
    },
    useSortBy
  );
```

Here we've added 3 things rather than just the one `disableSortRemove: true`. We've added:  
`disableSortRemove: true`,
`defaultCanSort: true`, 
and ` initialState: {
    sortBy: [{ id: tableData[0]?.totalSeries?.totalSeries, desc: true }] // IMPORTANT: use your own data shape here
 },`
 
 Not certain you need the second one (`defaultCanSort: true`) BUT the first and third give you the effect of, on page load, sorting is on and once you click the sort button it sorts the opposite way. @yyyyaaa

**Comment by RaviRamDhali** [comment]
ver8 this has changed to :  enableSortingRemoval
https://tanstack.com/table/v8/docs/api/features/sorting#enablesortingremoval

```
const table = useReactTable({
... 
... 
enableSortingRemoval: false,
})
```
